packages/smooth_app/lib/pages/product/add_basic_details_page.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/cards/product_cards/product_image_carousel.dart';
+import 'package:smooth_app/database/product_query.dart';
+import 'package:smooth_app/generic_lib/buttons/smooth_action_button.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/dialogs/smooth_alert_dialog.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_text_form_field.dart';
+
+class AddBasicDetailsPage extends StatefulWidget {
+  const AddBasicDetailsPage(this.product);
+  final Product product;
+  @override
+  State<AddBasicDetailsPage> createState() => _AddBasicDetailsPageState();
+}
+
+class _AddBasicDetailsPageState extends State<AddBasicDetailsPage> {
+  final TextEditingController _productNameController = TextEditingController();
+  final TextEditingController _brandNameController = TextEditingController();
+  final TextEditingController _weightController = TextEditingController();
+
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  @override
+  void initState() {
+    super.initState();
+    initializeProduct();
+  }
+
+  void initializeProduct() {
+    _productNameController.text = widget.product.productName ?? '';
+    _weightController.text = widget.product.quantity ?? '';
+    _brandNameController.text = widget.product.brands ?? '';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final Size size = MediaQuery.of(context).size;
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    return Scaffold(
+      appBar: AppBar(title: Text(appLocalizations.basic_details)),
+      body: Form(
+        key: _formKey,
+        child: ListView(
+          children: <Widget>[
+            Align(
+              alignment: Alignment.topLeft,
+              child: ProductImageCarousel(
+                widget.product,
+                height: size.height * 0.20,
+                onUpload: (_) {},
+              ),
+            ),
+            SizedBox(height: size.height * 0.02),
```

Comment:
I don't believe it's such a good idea to make it function of the screen height.
We have design_constants.dart to have consistent spacing in the whole app.
Besides, you repeat several times the same `SizedBox` height: please create a `final double` that contains the same height you use everywhere.

Reply:
Changes done, Created` final double _heightSpace`.